File: src/main/java/no/nav/syfo/domain/Inntektsmelding.java
package no.nav.syfo.domain;

import lombok.Builder;
import lombok.Getter;
import lombok.Value;

import java.util.List;
import java.util.Optional;

@Value
@Builder(toBuilder = true)
@Getter
public class Inntektsmelding {
    String fnr;
    Optional<String> arbeidsgiverOrgnummer;
    Optional<String> orgnummerPrivatperson;
    String journalpostId;
    String arbeidsforholdId;
    String arsakTilInnsending;
    String status;
    List<Periode> arbeidsgiverperioder;
}

File: src/main/kotlin/no/nav/syfo/domain/Inntektsmelding.kt
package no.nav.syfo.domain


data class Inntektsmelding(
    val fnr: String,
    val arbeidsgiverOrgnummer: String? = null,
    val arbeidsgiverPrivat: String? = null,
    val journalpostId: String,
    val arbeidsforholdId: String? = null,
    val arsakTilInnsending: String,
    val status: JournalStatus,
    val arbeidsgiverperioder: List<Periode> = emptyList()
)

